src/main/java/ua/olena/tracing/graph/Graph.java

```diff
@@ -0,0 +1,244 @@
+            }
+            //If I reached the node that has no path to destination
+            if (neighbors.isEmpty() && !node.equals(to)) {
+                goBack(visited, depthStack);
+            }
+            //I reached destination
+            else if (node.equals(to) && visited.size() > 1) {
+                int averageLatency = getAverage(visited);
+                if (minLatency == 0 || minLatency > averageLatency)
+                    minLatency = averageLatency;
+
+                goBack(visited, depthStack);
+            }
+        }
+
+        return minLatency;
+    }
+
+    /**
+     * Get all traces that are shorter than provided number between two services.
+     *
+     * @param from Node to start from
+     * @param to Destination node
+     * @param maxLatency maximal latency allowed
+     * @return Number of traces that are shorter than provided number
+     */
+    public Integer getShortestTraces(String from, String to, Integer maxLatency) {
+        int shortestTraces = 0;
+        int depth = 0;
+
+        Stack<String> visited = new Stack<>();
+        Stack<String> stack = new Stack<>();
+        Stack<Integer> depthStack = new Stack<>();
+        stack.push(from);
+        depthStack.push(depth);
+        while (!stack.isEmpty()) {
+            String node = stack.pop();
+            depth = depthStack.pop() + 1;
+            visited.add(node);
+            Vertex vertex = vertexes.get(node);
+            Map<String, Vertex> neighbors = vertex.getNeighbors();
+
+            int averageLatency = getAverage(visited);
+            if (averageLatency > maxLatency) {
+                goBack(visited, depthStack);
+            } else {
+                pushNeighborsToStack(depth, stack, depthStack, neighbors);
+
+                //If I reached the node that has no path to destination
+                if (neighbors.isEmpty() && !node.equals(to)) {
+                    goBack(visited, depthStack);
+                }
+                //I reached destination
+                else if (node.equals(to) && visited.size() > 1) {
+                    if (averageLatency < maxLatency)
+                        shortestTraces++;
+
+                    goBack(visited, depthStack);
+                }
+            }
+        }
+
+        return shortestTraces;
+    }
+
+    private void goBack(Stack<String> visited, Stack<Integer> depthStack) {
+        if (!depthStack.isEmpty()) {
+            int deltaDepth = depthStack.peek();
+            int backDepth = visited.size() - deltaDepth;
+            for (int i = 0; i < backDepth; i++)
+                visited.pop();
+        }
+    }
+
+    private int getAverage(Stack<String> visited) {
+        int average = 0;
+        Queue<String> queue = new ArrayDeque<>(visited);
+        String from = queue.poll();
+        while (!queue.isEmpty()) {
+            String to = queue.poll();
+            Edge edge = edges.get(from + to);
+            List<Integer> latencies = edge.getLatencies();
```

Comment:
see comment far above. there should be only one edge with one latency between two vertices to by best understanding.

Reply:
I did not understand your point here. I'm getting one edge and all the traces between these two services. What did I do wrong?